services/sources/publisher_backlist_service.py

```diff
@@ -158,41 +159,77 @@ def get_records(
-                    continue
-
-                s3_url = f'https://{bucket}.s3.amazonaws.com/{s3_path}'
                 
                 publisher_backlist_record = PublisherBacklistMapping(record_metadata)
                 publisher_backlist_record.applyMapping()
                 
-                webpub_flags=FileFlags(reader=True, limited_access=True)
-                self.add_has_part_mapping(s3_url, publisher_backlist_record.record, record_permissions['is_downloadable'], record_permissions['requires_login'])
-                self.s3_manager.store_pdf_manifest(publisher_backlist_record.record, self.file_bucket, flags=webpub_flags, path='/publisher_backlist')
+                try:
+                    file_url = self.download_file_from_location(record_metadata, record_permissions, publisher_backlist_record.record)
+                except Exception:
+                    logger.exception(f'Failed to download file for {record}')
+                    continue
                 
+                publisher_backlist_record.record.has_part.append(str(Part(
+                    index=1,
+                    url=file_url,
+                    source=publisher_backlist_record.record.source,
+                    file_type='application/pdf',
+                    flags=str(FileFlags(download=record_permissions['is_downloadable'], nypl_login=record_permissions['requires_login']))
+                )))
+                self.s3_manager.store_pdf_manifest(publisher_backlist_record.record, self.file_bucket, flags=FileFlags(reader=True, nypl_login=record_permissions['requires_login']), path='publisher_backlist')
                 mapped_records.append(publisher_backlist_record)
             except Exception:
                 logger.exception(f'Failed to process Publisher Backlist record: {record_metadata}')
         
         return mapped_records
+    
+    def download_file_from_location(self, record_metadata: dict, record_permissions: dict, record: Record) -> str:
+        file_location = record_metadata.get('DRB_File Location')
+
+        if not file_location:
+            hath_identifier = next((identifier for identifier in record.identifiers if identifier.endswith('hathi')), None)
+
+            if hath_identifier is not None:
+                hathi_id = hath_identifier.split('|')[0]
+
+                s3 = boto3.client("s3")
+
+                try:
+                    s3.head_object(Bucket=os.environ.get('PDF_BUCKET', None), Key=f'tagged_pdfs/{hathi_id}.pdf')
+                except Exception:
+                    logger.exception('PDF object does not exist')
+                    raise Exception
+
+                source_bucket_key = {'Bucket': os.environ.get('PDF_BUCKET', None), 'Key': f'tagged_pdfs/{hathi_id}.pdf'}
+                try:
+                    s3.copy(
+                        source_bucket_key,
+                        Bucket='drb-files-qa',
```

Comment:
should this be an env var so we can use this in prod? Can also be done in a separate PR!

Reply:
We actually have a env var in our config files for the DRB S3 Bucket called "FILE_BUCKET." I'll get the env var and replace the specific bucket name here with that instead.